openedx/core/djangoapps/programs/utils.py

```diff
@@ -111,6 +115,11 @@ def invert_programs(self):
 
         for program in self.programs:
             for course in program['courses']:
+                course_uuid = course['uuid']
+                if course_uuid in self.course_uuids:
+                    program_list = inverted_programs[course_uuid]
+                    if program not in program_list:
+                        program_list.append(program)
```

Comment:
Once the program is appended so we need to continue looking for enrollments in the runs?